core/build.gradle.kts

```diff
@@ -0,0 +1,122 @@
+plugins {
+    kotlin("android")
+    kotlin("kapt")
+    `maven-publish`
+
+    id("com.android.library")
+    id("com.google.devtools.ksp")
+}
+
+android {
+    compileSdk = 34
+
+    defaultConfig {
+        minSdk = 21
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+        consumerProguardFiles("consumer-rules.pro")
+    }
+
+    buildTypes {
+        release {
+            isMinifyEnabled = false
+            proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
+        }
+    }
+
+    buildFeatures {
+        dataBinding = true
+        viewBinding = true
+    }
+
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_17
+        targetCompatibility = JavaVersion.VERSION_17
+    }
+
+    kotlinOptions {
+        jvmTarget = "17"
+    }
+
+    namespace = "be.appwise.core"
+}
+
+dependencies {
+    implementation(libs.startup.runtime)
+
+    testImplementation(libs.mockk)
+    testImplementation(libs.junit)
+    androidTestImplementation(libs.junit.ext)
+    androidTestImplementation(libs.test.runner)
+    androidTestImplementation(libs.espresso.core)
+    api(libs.core.ktx)
+
+    api(libs.multidex)
+
+    //Kotlin reflect
+    api(libs.kotlin.reflect)
+
+    //ViewModel
+    api(libs.lifecycle.viewmodel.ktx)
+
+    api(libs.lifecycle.livedata.ktx)
+
+    //AndroidX Support libraries
+    api(libs.appcompat)
+    api(libs.material)
+    api(libs.constraintlayout)
+    api(libs.swiperefreshlayout)
+
+    //coroutines
+    api(libs.kotlinx.coroutines.core)
+    api(libs.kotlinx.coroutines.android)
+
+    // navigation components
+    api(libs.navigation.fragment.ktx)
+    api(libs.navigation.ui.ktx)
+
+    // gson (https://github.com/google/gson)
+    api(libs.gson)
+
+    // logger (https://github.com/orhanobut/logger)
+    api(libs.logger)
+
+    // Hawk (https://github.com/orhanobut/hawk)
+    api(libs.hawk)
+
+    //open image chooser
+    api(libs.easyimage)
+
+    //image loading/caching
+    api(libs.glide.glide)
+    ksp(libs.glide.compiler)
+
+    //dialogs
+    api(libs.material.dialogs.core)
+    api(libs.material.dialogs.datetime)
+
+    // Time manipulation for Java 7
+    api(libs.joda.time)
+
+    // https://github.com/Ereza/CustomActivityOnCrash
+    implementation(libs.customactivityoncrash)
+}
+
+val sourceJar: Task by tasks.creating(Jar::class) {
+    from(android.sourceSets["main"].java.srcDirs)
+//    classifier("sources")
+}
+
+afterEvaluate {
```

Comment:
Maybe an idea to move this to a separate file and include it where it is necessary? 
This could also be done with the submodules configuration with a filter on the specific module

Reply:
I tried this but it could not get it to work. I tried extracting it into an other publish.gradle.kts file and applying it with apply { from(../publish.gradle.kts) } but the publish.gradle.kts gave several errors regarding unresolved references like groupId etc